base_layer/core/src/blocks/blockheader.rs

```diff
@@ -118,11 +121,15 @@ impl BlockHeader {
             range_proof_mr: vec![0; 32],
             kernel_mr: vec![0; 32],
             total_kernel_offset: BlindingFactor::default(),
-            total_difficulty: Difficulty::default(),
             nonce: 0,
-            pow: TariProofOfWork::default(),
+            pow,
         }
     }
+
+    /// Calculates and returns the achieved difficulty for this header and associated proof of work.
+    pub fn achieved_difficulty(&self) -> Difficulty {
+        self.pow.achieved_difficulty(self)
+    }
 }
 
```

Comment:
should we not add a function to get achieved_difficulty() + prev.pow() as a single function?. I think this should make the code cleaner.

Reply:
We could